### PR TITLE
Git semver

### PR DESCRIPTION
**Pull request by bachp**

### CODE REVIEW

**Comment by png85** [comment]
LGTM - I've integrated this into my fork of this repository.

File: GetGitRevisionDescription.cmake [hunk 1]
```diff
@@ -84,38 +84,21 @@ function(git_describe _var)
 	if(NOT GIT_FOUND)
 		find_package(Git QUIET)
 	endif()
-	get_git_head_revision(refspec hash)
```
Comment (rpavlik) [inline GetGitRevisionDescription.cmake]: I think the reason I originally did this was to ensure that a git commit would trigger a re-CMake if desired. Does this behavior still work with your changes?

File: GetGitRevisionDescription.cmake [hunk 2]
```diff
@@ -84,38 +84,21 @@ function(git_describe _var)
...
 		describe
-		${hash}
 		${ARGN}
 		WORKING_DIRECTORY
 		"${CMAKE_CURRENT_SOURCE_DIR}"
 		RESULT_VARIABLE
 		res
 		OUTPUT_VARIABLE
 		out
-		ERROR_QUIET
```
Comment (rpavlik) [inline GetGitRevisionDescription.cmake]: Any reason why this line got removed?

**Comment by rpavlik** [comment]
Sorry for taking so long to get to this - GitHub notifications are a wall of noise to me at this point. I think it's good in principle (might even fix stuff like running in a submodule with the new-style setup where `.git` is a file instead of directory, with the object store in the outer module's `.git` dir), just have some questions before merging fairly substantial changes to a widely-used module's core functionality.

**Comment by bachp** [comment]
I need to dive in to this changes again. I don't really remember the details :p. I will come back to you once I'm able to answer.

**Pull request closed without merging**